spec/mongoid/copyable_spec.rb

```diff
@@ -696,5 +696,455 @@
         end
       end
     end
+
+    context "when fields are removed before cloning" do
+
+      context "when using embeds_one associations" do
+
+        before do
+          class CloneParent
+            include Mongoid::Document
+
+            embeds_one :clone_child
+
+            field :a, type: :string
+            field :b, type: :string
+          end
+
+          class CloneChild
+            include Mongoid::Document
+
+            embedded_in :clone_parent
+            embeds_one :clone_grandchild
+
+            field :c, type: :string
+            field :d, type: :string
+          end
+
+          class CloneGrandchild
+            include Mongoid::Document
+
+            embedded_in :clone_child
+
+            field :e, type: :string
+            field :f, type: :string
+          end
+
+        end
+
+        after do
+          Object.send(:remove_const, :CloneParent)
+          Object.send(:remove_const, :CloneChild)
+          Object.send(:remove_const, :CloneGrandchild)
```

Comment:
Is this copy-pasted from elsewhere? I would prefer to use permanently defined models in spec/support/models etc.